tests/replication/tcp/streams/test_typing.py

```diff
@@ -43,8 +44,8 @@ def test_typing(self) -> None:
         request = self.handle_http_replication_attempt()
         self.assert_request_is_get_repl_stream_updates(request, "typing")
 
-        self.test_handler.on_rdata.assert_called_once()
-        stream_name, _, token, rdata_rows = self.test_handler.on_rdata.call_args[0]
+        self.mock_handler.on_rdata.assert_called_once()
+        stream_name, _, token, rdata_rows = self.mock_handler.on_rdata.call_args[0]
```

Comment:
I'm guessing `test_handler` was (and still is) defined by `BaseStreamTestCase`. Why do we want a separate `mock_handler` here?

Reply:
Because the type is different on `BaseStreamTestCase` (it is `TestReplicationDataHandler`), which doesn't have all the `mock` properties. I could maybe make the classes generic over `_build_replication_data_handler` or something, but that seems like overkill for tests?

Reply:
Absolutely overkill, just wanted to understand the reasoning.